### PR TITLE
Tidy 404 Not Found error page

### PR DESCRIPTION
**Pull request by sftim**
*Note* if you find any non-blocking defects whilst reviewing these changes, those could well make Good First Issues.

- [current](https://k8s.io/foobar)
- [proposed](https://deploy-preview-23753--kubernetes-io-master-staging.netlify.app/foobar)

### CODE REVIEW

File: layouts/404.html [hunk 1]
```diff
@@ -1,29 +1,14 @@
...
-            </a>
-        </li>
+<section id="error-404" class="error-details">
+    <p>
+    {{ T "error_404_were_you_looking_for" }}
+    </p>
+    <ul id="error-sections">
+        {{ $sections := slice "docs" "blog" "training" "partners" "community" "case-studies" }}
+        {{ range $sections }}
+        {{ with site.GetPage "section" . }}<li><a href="{{ .RelPermalink }}" data-proofer-ignore>{{ .Title }}</a></li>{{ end }}
```
Comment (kbhawkey) [inline layouts/404.html:10]: Thanks @sftim. These changes look nice.
One question: At this time, is there a good reason to add "data-proofer-ignore"?

**Comment by sftim** [comment]
I might well revisit https://github.com/kubernetes/website/pull/23753#discussion_r486363764, probably after this merges if that's OK?

**Comment by kbhawkey** [comment]
The changes improve the current "404" page.
/approve

**Pull request merged**